### PR TITLE
docs(config, world/upgrade): add proxy support

### PR DESCRIPTION
**Pull request by qbzzt**
MUD 2.0.7 and above support this function. Ready to merge.

### CODE REVIEW

File: docs/pages/config.mdx [hunk 1]
```diff
@@ -121,3 +121,7 @@ The global configuration keys are all optional.
 
   - **`worldsFile`** a `string`: JSON file for the chain to `World` deploy address mapping.
     The default is `./worlds.json`.
+
+  - <a id="upgradeableWorldImplementation" />
+    **`upgradeableWorldImplementation`** a `bool`: Whether the `World` is to be deployed behind a proxy to [enable upgrades](/world/upgrade).
```
Comment (alvrs) [inline docs/pages/config.mdx:126]:
```suggestion
    **`upgradeableWorldImplementation`** a `bool`: Whether the `World` is to be deployed behind a proxy to [enable upgrades of the core World implementation](/world/upgrade).
```

File: docs/pages/world/upgrade.mdx [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { CollapseCode } from "../../components/CollapseCode";
+import { Callout } from "nextra/components";
+
+# Upgrading worlds
+
+The [tables](./tables) and [`System`s](./systems) can be upgraded without changing the underlying `World` code.
```
Comment (alvrs) [inline docs/pages/world/upgrade.mdx:6]:
```suggestion
The [`System`s](./systems) can be upgraded without changing the underlying `World` code.
```
Let's not talk about upgrading systems before we have support for it

File: docs/pages/config.mdx (hunk 1 above)
Comment (holic) [inline docs/pages/config.mdx:126]: Worlds are technically already upgradeable, but only via tables and systems.

We should clarify that this upgrades the _core implementation_ of a world, the things that can't be upgraded through tables/systems, and the things that may get security patches in the future.

File: docs/pages/world/upgrade.mdx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+However, if the `World` was deployed [behind a proxy](/config#upgradeableWorldImplementation) it is also possible to upgrade the `World` itself.
+This will be useful in the future as we further develop MUD, adding features and fixing bugs.
```
Comment (alvrs) [inline docs/pages/world/upgrade.mdx:8]:
```suggestion
This allows you to upgrade to a future version if MUD, but adds some gas overhead for all calls (due to one more level of indirection).
```

File: docs/pages/world/upgrade.mdx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+## Are upgrades possible?
+
+As per [ERC-1967](https://eips.ethereum.org/EIPS/eip-1967#logic-contract-address), the storage slot `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc` stores the address of the actual `World` contract in a proxy.
```
Comment (alvrs) [inline docs/pages/world/upgrade.mdx:12]:
```suggestion
As per [ERC-1967](https://eips.ethereum.org/EIPS/eip-1967#logic-contract-address), the storage slot `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc` stores the address of the `World` implementation in a proxy.
```

File: docs/pages/world/upgrade.mdx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+So to identify if a `World` is deployed behind a proxy run these commands:
+
+```sh copy
+WORLD_ADDRESS=0xbe6b85dc88f969e45d8d8ae128c5a9c9744d6464
+cast implementation $WORLD_ADDRESS
+```
+
+If the answer is anything other than zero, the `World` is behind a proxy and can be upgraded.
+
+## The actual upgrade
```
Comment (holic) [inline docs/pages/world/upgrade.mdx:22]:
```suggestion
## Performing an upgrade
```

File: docs/pages/world/upgrade.mdx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+   contract SetImplementation is Script {
+     function run() external {
+       uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+       vm.startBroadcast(deployerPrivateKey);
+
+       address proxyAddress = vm.envAddress("WORLD_ADDRESS");
+
+       // Deploy new world implementation
+       World newWorld = new World();
```
Comment (holic) [inline docs/pages/world/upgrade.mdx:70]: can we make a note somewhere to revisit this in a couple weeks and wire this up to the deterministic deployer?

Comment (qbzzt) [reply docs/pages/world/upgrade.mdx:70]: Mind if I do it for mid May? Things always take longer than expected.

Comment (qbzzt) [reply docs/pages/world/upgrade.mdx:70]: Event created.

File: docs/pages/world/upgrade.mdx [hunk 7]
```diff
@@ -0,0 +1,95 @@
+import { CollapseCode } from "../../components/CollapseCode";
+import { Callout } from "nextra/components";
+
+# Upgrading worlds
+
+The [`System`s](./systems) can be upgraded without changing the underlying `World` code.
+However, if the `World` was deployed [behind a proxy](/config#upgradeableWorldImplementation) it is also possible to upgrade the `World` itself.
+This allows you to upgrade to a future version if MUD, but adds some gas overhead for all calls (due to one more level of indirection).
```
Comment (yonadaaa) [inline docs/pages/world/upgrade.mdx:8]:
```suggestion
This allows you to upgrade to a future version of MUD, but adds some gas overhead for all calls (due to one more level of indirection).
```

**Review by yonadaaa — CHANGES_REQUESTED** [review]

**Review by yonadaaa — APPROVED** [review]

**Pull request merged**